docs/installation/microcontrollers/_index.md

```diff
@@ -293,9 +254,75 @@ In this example, we want to add GPIO pin 35, which is labeled `ADC1_7` in the pi
             ];
     ```
 
-## Next Steps
+{{% /expand%}}
+
+### Upload the Project and Connect to your ESP32 Board
+
+After modifying the contents of <file>src/main.rs</file> to your liking, run:
+
+``` shell
+make upload
+```
+
+While running `make upload`, you may be presented with an interactive menu of different serial port options to use to connect to the ESP32 board.
+Once you have identified the correct choice for your environment, you may bypass the menu by providing the correct port as a command line argument to future invocations of `make upload`:
+
+``` shell
+make ESPFLASH_FLASH_ARGS="-p /dev/cu.usbserial-130" upload
+```
+
+If successful, `make upload` will retain a serial connection to the board until `Ctrl-C` is pressed.
+To manage this connection, consider running it within a dedicated terminal session, or under `tmux` or `screen`.
+While the serial connection is live, you can also restart the currently flashed image with `Ctrl-R`.
+
+If everything went well, your ESP32 will be programmed so that you will be able to see your robot live on [the Viam app](https://app.viam.com).
+
+{{< alert title="Note" color="note" >}}
+
+If you encounter a crash due to stack overflow, you may need to increase the stack available to the main task.
+
+In the generated <file>sdkconfig.defaults</file> set the `CONFIG_ESP_MAIN_TASK_STACK_SIZE` to `32768`. The diff of your changes should look like this:
+
+``` diff
+diff --git a/sdkconfig.defaults b/sdkconfig.defaults
+index f75b465..2b0ba9c 100644
+--- a/sdkconfig.defaults
++++ b/sdkconfig.defaults
+@@ -1,5 +1,5 @@
+ # Rust often needs a bit of an extra main task stack size compared to C (the default is 3K)
+-CONFIG_ESP_MAIN_TASK_STACK_SIZE=24576
++CONFIG_ESP_MAIN_TASK_STACK_SIZE=32768
+ CONFIG_ESP_MAIN_TASK_AFFINITY_CPU1=y
+ # Use this to set FreeRTOS kernel tick frequency to 1000 Hz (100 Hz by default).
+```
+
+{{% /alert %}}
+
+## Configure the controlling robot
+
+You have already created one robot which is running the Micro-RDK on your ESP32. To programmatically control the robot now running on the ESP32, you need to connect it to another robot that is running the full-featured `viam-server` software, as the microcontroller lacks the required processing power to do so.
+This second "robot" can be as simple as an instance of `viam-server` running on your development machine.
+
+### Configure the ESP32 as a Remote
+
+Navigate to [the Viam app](https://app.viam.com).
+Create and configure a second new robot.
+
+Add the ESP32-backed robot as a remote of your new second robot to establish a secure connection between both robots:
+
+<div style="max-width:800px;"><img src="../img/esp32-setup/esp32-remote-creation.png" alt="Adding the ESP32 as a remote in the Viam app Config builder." ></div>
+
+1. Navigate to the **CONTROL** tab of the robot and copy its `Remote Address`.

```

Comment:
1. make it clear that they're doing this on the worker robot.
2. @npmenard is the control tab really the best place for this? can they get this from the code tab? my concern is that the control tab will error out at this point.

Reply:
yes you are right they should use the Code sample tab and copy paste what's in the remote section